uvicorn/protocols/websockets/websockets_impl.py

```diff
@@ -145,7 +144,8 @@ def connection_lost(self, exc: Optional[Exception]) -> None:
 
     def shutdown(self) -> None:
         self.ws_server.closing = True
-        self.transport.close()
+        if not self.transport.is_closing():
```

Comment:
Do I need to check if the transport is closing? I don't think so...

Do I need to send 500 if handshake is not completed?

Reply:
No, I don't need to check if the transport is closing.

Yes, I do need to send 500 if handshake is not completed.